Add `close_tab` keybinding action for macOS

liby opened the pull request:
This PR adds support for a dedicated `close_tab` keybinding action, allowing users to bind specific keys for closing tabs. The implementation:

- Adds `close_tab` as a new keybinding action
- Preserves all existing confirmation dialogs for running processes
- Works seamlessly with macOS native tab system

### Testing
- [x] Tested with single tabs
- [x] Tested with multiple tabs
- [x] Tested with running processes (confirmation dialog)
- [x] Tested with splits within tabs

<img width="797" alt="image" src="https://github.com/user-attachments/assets/8e09eea3-1f71-40a3-a835-76de14013a29" />

https://github.com/user-attachments/assets/155210f7-20fe-4a96-8800-6969df214871

Partially resolved #4331

mitchellh commented:
> Resolves https://github.com/ghostty-org/ghostty/issues/4331

Just noting this is not true, since we have to also implement it for GTK. But I think there's a PR for that too.

Thanks for doing this, I'll review soon!

liby committed:
Add `close_tab` keybinding action for macOS

Implement `close_tab` keybinding action to close the current tab and all splits within that tab.

mitchellh commented:
Awesome work! I made some minor modifications to fit in with the GTK work done. Auto-merge is on and then this is supported on both platforms.
